api/ondemand_vhds_rds_protocol.rst

```diff
@@ -0,0 +1,187 @@
+.. _on_demand_vhds_rds_protocol:
+
+On-Demand RDS/VHDS
+==================
+Currently, in RDS, all routes for the cluster are sent to every Envoy instance
+in the mesh. This causes scaling issues as the size of the cluster grows. The
+majority of this complexity can be found in the virtual host configurations, of
+which most are not needed by any individual proxy. With a goal of being able
+to scale to one million vhosts in the future, the capability to filter a
+proxy to only contain the required virtual hosts is a necessity.
+
+In order to fix this issue, we are implementing on-demand RDS. On-demand RDS,
```

Comment:
Isn't this on-demand VHDS, and we need to motivate the fact that VHDS has been added as an extra indirection layer , to providing the resource addressing granularity needed to do on-demand?

Reply:
cleaned up. Thanks.